.github/workflows/main.yml

```diff
@@ -201,13 +219,28 @@ jobs:
         if: failure() && steps.run_tests.conclusion == 'failure'
         uses: actions/upload-artifact@v4
         with:
-          name: rest_api_container_logs
+          name: ${{
+            contains(matrix.pattern, 'not')
+              && 'rest_api_container_logs'
+              || format(
+                  'rest_api_container_logs-{0}-cache-{1}',
+                  matrix.pattern, matrix.allow_cache
+                )
+            }}
           path: "${{ github.workspace }}/rest_api_testing"
+
       - name: Upload allure results
         if: always()
         uses: actions/upload-artifact@v4
         with:
-          name: allure-results-rest-api
+          name: ${{
+            contains(matrix.pattern, 'not')
```

Comment:
IMO this pattern is unnecessary complicated. Basically the main issue is that in any case we end up having too many artifacts 
![image](https://github.com/user-attachments/assets/0e800f28-df64-4254-9ccb-135e65163cb3)
I'd suggest using merge action https://github.com/actions/upload-artifact/blob/main/merge/README.md. If you feel that it can be achieved in this PR, please do it. Otherwise we will create PR.

But in any case please simplify the filter. Smth like this would be enough 
`allure-results-tests-api-${{matrix.pattern}}-cache-${{matrix.allow_cache}}`